modules/date-picker/js/date-picker_directive.js

```diff
@@ -123,7 +123,7 @@
 
             timer2 = $timeout(function()
             {
-                var yearSelector = $element.find('.lx-date-picker__year-selector');
+                var yearSelector = $document.find('.lx-date-picker__year-selector');
```

Comment:
Hi @poiswe I also came across this issue today and have a suggestion.  You can already access the relevant modal's element via lxDatePicker.element, as opposed to scanning the entire $document.

var yearSelector = lxDatePicker.element.find('.lx-date-picker__year-selector');